kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -1028,16 +1028,14 @@ protected CompletableFuture<Void> storeGroupId(String groupId, String groupIdPat
     @VisibleForTesting
     public <T> void replaceTopicPartition(Map<TopicPartition, T> replacedMap,
                                           Map<TopicPartition, TopicPartition> replacingIndex) {
-        String namespacePrefix = currentNamespacePrefix();
         Map<TopicPartition, T> newMap = new HashMap<>();
         replacedMap.entrySet().removeIf(entry -> {
             if (replacingIndex.containsKey(entry.getKey())) {
                 newMap.put(replacingIndex.get(entry.getKey()), entry.getValue());
                 return true;
             } else if (KopTopic.isFullTopicName(entry.getKey().topic())) {
                 newMap.put(new TopicPartition(
-                                KopTopic.removeDefaultNamespacePrefix(entry.getKey().topic(),
-                                        namespacePrefix),
+                                KopTopic.removePersistentDomain(entry.getKey().topic()),
```

Comment:
What if there are two applications that one uses `public/default/my-topic` and the other uses `my-topic`? Do they share the same offset after that? If not, it might cause unexpected changes. I think we might need to add the namespace prefix for the short topic name.

Reply:
Ignore my comments. I've verified it.